clustercontroller-utils/src/main/java/com/yahoo/vespa/clustercontroller/utils/staterestapi/server/JsonReader.java

```diff
@@ -98,7 +109,33 @@ public SetRequestData getStateRequestData(HttpRequest request) throws Exception
             }
             stateMap.put(type, new UnitStateImpl(code, reason));
         }
-        return new SetRequestData(stateMap, condition, responseWait);
+
+        final Optional<Duration> timeout = parseTimeout(request.getOption("timeout", null));
+
+        return new SetRequestData(stateMap, condition, responseWait, timeout);
+    }
+
+    public static Optional<Duration> parseTimeout(String timeoutOption) throws InvalidContentException {
+        if (timeoutOption == null) {
+            return Optional.empty();
+        } else {
+            float timeoutSeconds;
+            try {
+                timeoutSeconds = Float.parseFloat(timeoutOption);
+            } catch (NumberFormatException e) {
+                throw new InvalidContentException("value of timeout->" + timeoutOption + " is not a float");
+            }
+
+            if (timeoutSeconds <= 0.0) {
+                return Optional.of(Duration.ZERO);
+            } else if (timeoutSeconds <= MAX_TIMEOUT.getSeconds()) {
+                long micros = Math.round(timeoutSeconds * MICROS_IN_SECOND);
+                long nanoAdjustment = TimeUnit.MILLISECONDS.toNanos(micros % MICROS_IN_SECOND);
+                return Optional.of(Duration.ofSeconds(micros / MICROS_IN_SECOND, nanoAdjustment));
```

Comment:
Could we simplify to e.g. `Duration.ofMillis((long)(timeoutSeconds * MILLIS_IN_SECOND))` and let the implementation deal with the nanosecond extraction? I don't think we'll need more than millisecond resolution for this.

Reply:
Done